execsrc/Main.hs

```diff
@@ -53,39 +73,45 @@ argumentParser = info (arguments <**> helper)
                  )
 
 -- | The actual entry point of the application
-application :: FilePath -> FilePath -> EvalM ()
-application ss dirOfModelSolutions = do
+application :: Gen String -> FilePath -> FilePath -> EvalM ()
+application gen ss dirOfModelSolutions = do
```

Comment:
I would rather make an issue of this. This file is probably going to be changing a lot in the future anyway, right now we need to get on to strategies. The best refactoring is not obvious.